src/analysis/calculus/times_cont_diff.lean

```diff
@@ -14,591 +14,1162 @@ By induction, it is `C^n` if it is `C^{n-1}` and its (n-1)-th derivative is `C^1
+it only works well if all spaces belong to the same universe. To get the general version, we lift
+things to a common universe using a trick.
+
+### Variables management
+
+The mathbook definitions and proofs use various identifications and abuse of notations, for instance
+when saying that the natural space in which the derivative lives, i.e.,
+`E →L[𝕜] (E →L[𝕜] ( ... →L[𝕜] F))`, is the same as a space of multilinear maps. When doing things
+formally, we need to provide explicit maps for these identifications, and chase some diagrams to see
+everything is compatible with the identifications. In particular, one needs to check that taking the
+derivative and then doing the identification, or first doing the identification and then taking the
+derivative, gives the same result. The key point for this is that taking the derivative commutes
+with continuous linear equivalences. Therefore, we need to implement all our identifications with
+continuous linear equivs.
+
+## Notations
+
+We write `E [×n]→L[𝕜] F` for the space of continuous multilinear maps on `E^n` with values in `F`.
+This is the space in which the `n`-th derivative of a function from `E` to `F` lives.
 
 ## Tags
 
-derivative, differentiability, higher derivative, C^n
-
+derivative, differentiability, higher derivative, `C^n`, multilinear, Taylor series, formal series
 -/
 
 noncomputable theory
-local attribute [instance, priority 10] classical.decidable_inhabited classical.prop_decidable
+open_locale classical
 
 universes u v w
 
-open set
+open set fin
 open_locale topological_space
 
 variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
-{E : Type u} [normed_group E] [normed_space 𝕜 E]
-{F : Type u} [normed_group F] [normed_space 𝕜 F]
-{G : Type u} [normed_group G] [normed_space 𝕜 G]
-{s s₁ u : set E} {f f₁ : E → F} {f' f₁' : E →L[𝕜] F} {f₂ : E → G}
-{f₂' : E →L[𝕜] G} {g : F → G} {x : E} {c : F}
-{L : filter E} {t : set F} {b : E × F → G} {sb : set (E × F)} {p : E × F}
-{n : ℕ}
+{E : Type*} [normed_group E] [normed_space 𝕜 E]
+{F : Type*} [normed_group F] [normed_space 𝕜 F]
+{G : Type*} [normed_group G] [normed_space 𝕜 G]
+{s s₁ t u : set E} {f f₁ : E → F} {g : F → G} {x : E} {c : F}
+{b : E × F → G}
 
-include 𝕜
+set_option class.instance_max_depth 370
 
-/--
-The space `iterated_continuous_linear_map 𝕜 n E F` is the space E →L[𝕜] (E →L[𝕜] (E ... F)...))),
-defined inductively over `n`. This is the space to which the `n`-th derivative of a function
-naturally belongs. It is only defined when `E` and `F` live in the same universe.
--/
-def iterated_continuous_linear_map (𝕜 : Type w) [nondiscrete_normed_field 𝕜] :
-  Π (n : ℕ) (E : Type u) [gE : normed_group E] [@normed_space 𝕜 E _ gE]
-    (F : Type u) [gF : normed_group F] [@normed_space 𝕜 F _ gF], Type u
-| 0     E _ _ F _ _ := F
-| (n+1) E _ _ F _ _ := by { resetI, exact iterated_continuous_linear_map n E (E →L[𝕜] F) }
+notation E `[×`:25 n `]→L[`:25 𝕜 `] ` F := continuous_multilinear_map 𝕜 (λ (i : fin n), E) F
```

Comment:
Why not introduce this notation near the definition of `continuous_multilinear_map`?

Reply:
Done.